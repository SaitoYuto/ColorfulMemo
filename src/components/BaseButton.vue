<template>
  <v-btn
    :color="color"
    :text="text"
    :variant="variant"
    v-show="visibility"
  ></v-btn>
</template>

<script lang="ts" setup>
import { BUTTON_VARIANT } from "@/constants/ButtonVariant";
import type { ButtonVariant } from "@/types/ButtonVariant";
defineProps({
  color: {
    type: String,
    default: undefined,
  },
  text: {
    type: String,
    required: true,
  },
  variant: {
    type: String as PropType<ButtonVariant>,
    default: BUTTON_VARIANT.TONAL,
  },
  visibility: {
    type: Boolean,
    default: true,
  },
});
</script>
