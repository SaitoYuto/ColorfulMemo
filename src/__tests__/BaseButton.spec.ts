import BaseButton from "@/components/BaseButton.vue";
import { mount } from "@vue/test-utils";
import "@/__tests__/setup";

describe("BaseButton.vue", () => {
  it("renders props.text when passed", async () => {
    const text = "Button Text";
    const wrapper = mount(BaseButton, {
      props: {
        text: text,
      },
    });
    expect(wrapper.props("text")).toBe(text);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders props.text when an empty string passed", async () => {
    const emptyText = "";
    const wrapper = mount(BaseButton, {
      props: {
        text: emptyText,
      },
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.props("text")).toBe(emptyText);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders props.visibility when true passed", async () => {
    const text = "Button Text";
    const visible = true;
    const wrapper = mount(BaseButton, {
      props: {
        text: text,
        visibility: visible,
      },
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.isVisible()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("becomes invisible props.visibility when false passed", async () => {
    const text = "Button Text";
    const invisible = false;
    const wrapper = mount(BaseButton, {
      props: {
        text: text,
        visibility: invisible,
      },
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.isVisible()).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
