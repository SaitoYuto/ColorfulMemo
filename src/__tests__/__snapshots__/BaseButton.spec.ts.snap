// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButton.vue > becomes invisible props.visibility when false passed 1`] = `
"<button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="display: none;"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
  <!----><span class="v-btn__content" data-no-activator="">Button Text</span>
  <!---->
  <!---->
</button>"
`;

exports[`BaseButton.vue > renders props.isShow when true passed 1`] = `
"<button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" isshow="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
  <!----><span class="v-btn__content" data-no-activator="">Button Text</span>
  <!---->
  <!---->
</button>"
`;

exports[`BaseButton.vue > renders props.text when an empty string passed 1`] = `
"<button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
  <!----><span class="v-btn__content" data-no-activator=""></span>
  <!---->
  <!---->
</button>"
`;

exports[`BaseButton.vue > renders props.text when passed 1`] = `
"<button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
  <!----><span class="v-btn__content" data-no-activator="">Button Text</span>
  <!---->
  <!---->
</button>"
`;

exports[`BaseButton.vue > renders props.visibility when true passed 1`] = `
"<button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
  <!----><span class="v-btn__content" data-no-activator="">Button Text</span>
  <!---->
  <!---->
</button>"
`;
